skills_taxonomy_v2/pipeline/skills_extraction/skills_naming_utils.py

```diff
@@ -88,81 +123,44 @@ def clean_cluster_description(sentences):
         cluster_descriptions (dict): Cluster number : list of cleaned
             sentences in this cluster
     """
-    # Init the Wordnet Lemmatizer
-    lemmatizer = WordNetLemmatizer()
 
     # How many times a n-gram has to occur in order for occurences of them
     # to be converted to a single dash separated word
     num_times_ngrams_thresh = 3
 
-    work_stopwords = [
-        "essential",
-        "requirement",
-        "required" "degree",
-        "responsibility",
-        "duties",
-        "responsibilities",
-        "experienced",
-        "previous",
-        "andor",
-        "minimum",
-        "year",
-        "skill",
-        "ideal",
-        "candidate",
-        "desirable",
-        "willing",
-        "prepared",
-        "knowledge",
-        "experience",
-        "skills",
-        "ideally",
-        "responsible",
-        "require",
-        "environment",
-        "role",
-        "work",
-        "job",
-        "description",
-        "ymy",
-    ]
-
-    all_stopwords = stopwords.words("english") + work_stopwords
-
-    cluster_docs_cleaned = []
+    # Don't append duplicates, just use set
+    cluster_docs_cleaned = set()
     for sentence in sentences:
-        acronyms = re.findall("[A-Z]{2,}", sentence)
-        # Lemmatize
-        lemmatized_output = [
-            lemmatizer.lemmatize(w)
-            if w in acronyms
-            else lemmatizer.lemmatize(w.lower())
-            for w in sentence.split(" ")
-        ]
+        # acronyms = re.findall("[A-Z]{2,}", sentence)
+        # # Lemmatize - lemmatized_output not used
+        # lemmatized_output = [
+        #     lemmatizer.lemmatize(w)
+        #     if w in acronyms
+        #     else lemmatizer.lemmatize(w.lower())
+        #     for w in sentence.split(" ")
+        # ]
         # singularise
         singularised_output = [singularize(w) for w in sentence.split(" ")]
-        no_stopwords = [
-            word for word in singularised_output if word not in all_stopwords
+        no_numbers = [
+            word for word in singularised_output if ((word not in all_stopwords) and (not word.isdigit()))
         ]
-        no_numbers = [word for word in no_stopwords if not word.isdigit()]
-        cluster_docs_cleaned.append(" ".join(no_numbers))
+        cluster_docs_cleaned.add(" ".join(no_numbers))
 
-        # Remove duplicates
-        cluster_docs_cleaned = list(set(cluster_docs_cleaned))
+    cluster_docs_cleaned = list(cluster_docs_cleaned)
```

Comment:
`cluster_docs_cleaned = list(set(cluster_docs_cleaned))`

Reply:
`cluster_docs_cleaned` is a set already thats being added to, so no duplicates can get added to it in the first place